### PR TITLE
[WIP] Do Not Take Arguments for --shared Flag (Go CLI)

### PR DESCRIPTION
**Pull request by dubeejw**
- Remove logic for --shared flag to take arguments
- If --shared is present in a command, the entity will be public.
  Otherwise, the entity will not be publically shared

### CODE REVIEW

**Comment by dubeejw** [comment]
@mdeuser, please review.

**Comment by dubeejw** [comment]
IPG 222

**Comment by dubeejw** [comment]
@mdeuser, please review.

File: tools/go-cli/go-whisk-cli/commands/action.go [hunk 1]
```diff
@@ -720,7 +709,7 @@ func init() {
     actionCreateCmd.Flags().BoolVar(&flags.action.copy, "copy", false, "treat artifact as the name of an existing action")
     actionCreateCmd.Flags().BoolVar(&flags.action.sequence, "sequence", false, "treat artifact as comma separated sequence of actions to invoke")
     actionCreateCmd.Flags().StringVar(&flags.action.kind, "kind", "", "the kind of the action runtime (example: swift:3)")
-    actionCreateCmd.Flags().StringVar(&flags.action.shared, "shared", "", "shared action (default: private)")
+    actionCreateCmd.Flags().BoolVar(&flags.common.shared, "shared", false, "make an action publically accessible")
```
Comment (rabbah) [inline tools/go-cli/go-whisk-cli/commands/action.go]: publically -> publicly

File: tools/go-cli/go-whisk-cli/commands/action.go [hunk 2]
```diff
@@ -731,7 +720,7 @@ func init() {
     actionUpdateCmd.Flags().BoolVar(&flags.action.docker, "docker", false, "treat artifact as docker image path on dockerhub")
     actionUpdateCmd.Flags().BoolVar(&flags.action.copy, "copy", false, "treat artifact as the name of an existing action")
     actionUpdateCmd.Flags().BoolVar(&flags.action.sequence, "sequence", false, "treat artifact as comma separated sequence of actions to invoke")
-    actionUpdateCmd.Flags().StringVar(&flags.action.shared, "shared", "", "shared action (default: private)")
+    actionUpdateCmd.Flags().BoolVar(&flags.common.shared, "shared", false, "make an action publically accessible")
```
Comment (rabbah) [inline tools/go-cli/go-whisk-cli/commands/action.go]: publically -> publicly

File: tests/src/common/WskCli.java [hunk 3]
```diff
@@ -367,7 +367,7 @@ public String createAction(int expectedCode, String name, String artifact, Strin
         }
 
         if (shared) {
-            cmd = Util.concat(cmd, new String[] { "--shared", "yes"});
+            cmd = Util.concat(cmd, new String[] { "--shared" });
```
Comment (rabbah) [inline tests/src/common/WskCli.java]: missed removing the "yes"/"no" from Wsk.scala.

File: tools/go-cli/go-whisk-cli/commands/package.go [hunk 4]
```diff
@@ -185,16 +184,6 @@ var packageCreateCmd = &cobra.Command{
         }
         client.Namespace = qName.namespace
 
-        if (flags.common.shared == "yes") {
```
Comment (rabbah) [inline tools/go-cli/go-whisk-cli/commands/package.go]: I had understood the plan as making yes/no optional. I believe with this change, once you make the package public (or other types), you cannot make it private. Is that a correct statement?

Note that the python CLI allows yes/no optionally but requires --shared no to make the package private.
To wit, we need a test that makes a public package private via CLI.

Comment (dubeejw) [inline tools/go-cli/go-whisk-cli/commands/package.go]: The package we are using to parse arguments does not allow flags to to arguments, and not take arguments. When --shared is not present in a command, the entity will be set to not be shared.

Comment (dubeejw) [inline tools/go-cli/go-whisk-cli/commands/package.go]: shared

**Comment by rabbah** [comment]
So I create a package and make it public. Then I update the package and don't provide --shared, it becomes private?

**Comment by dubeejw** [comment]
Exactly. I can close this PR, and leave it the way it is to accept arguments.

**Comment by rabbah** [comment]
I'm inclined to say yes - leave as is. Let others weigh in.

File: tools/go-cli/go-whisk-cli/commands/package.go (hunk 4 above)
Comment (mdeuser) [inline tools/go-cli/go-whisk-cli/commands/package.go]: @rabbah - Do you think going forward this shared argument will eventually take additional states besides today's binary public|private values?  IMHO, I think we'll see additional shared scopes, including custom ones, in the future.  If so, then perhaps we should move this argument away from being a true|false state and more of a select-a-shared-scoped kind of argument.

File: tools/go-cli/go-whisk-cli/commands/action.go (hunk 1 above)
Comment (mdeuser) [inline tools/go-cli/go-whisk-cli/commands/action.go]: maybe also add what the default value is.. something like `(default: action is private)`.. maybe shorter `(default: private)`

File: tools/go-cli/go-whisk-cli/commands/package.go [hunk 5]
```diff
@@ -598,19 +550,19 @@ func init() {
     packageCreateCmd.Flags().StringSliceVarP(&flags.common.annotation, "annotation", "a", []string{}, "annotations")
     packageCreateCmd.Flags().StringSliceVarP(&flags.common.param, "param", "p", []string{}, "default parameters")
     packageCreateCmd.Flags().StringVarP(&flags.xPackage.serviceGUID, "service_guid", "s", "", "a unique identifier of the service")
-    packageCreateCmd.Flags().StringVar(&flags.common.shared, "shared", "" , "shared action (default: private)")
+    packageCreateCmd.Flags().BoolVar(&flags.common.shared, "shared", false, "make a package publically accessible")
```
Comment (mdeuser) [inline tools/go-cli/go-whisk-cli/commands/package.go]: "make a package.." -> "make package...(default: private)" (remove the 'a' indefinite article).  Same comment for other `shared` help text

**Comment by mdeuser** [comment]
Should the whisk/trigger.go have the following update similar to the action, package and rule modules?
In the `Trigger` struct used to encode the JSON sent to the server:
```
Publish bool `json:"publish,omitempty"`
```
change to

```
Publish bool `json:"publish"`  // remove the omitempty
```

File: tools/go-cli/go-whisk-cli/commands/package.go (hunk 4 above)
Comment (rabbah) [inline tools/go-cli/go-whisk-cli/commands/package.go]: fine grained entitlement is a feature we want to surface, eventually.

**Pull request closed without merging**